src/pudl/transform/eia930.py

```diff
@@ -0,0 +1,224 @@
+"""Module to perform data cleaning functions on EIA930 data tables."""
+
+import pandas as pd
+from dagster import AssetOut, Output, multi_asset
+
+import pudl
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+@multi_asset(
+    outs={
+        "core_eia930__hourly_balancing_authority_net_generation": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+        "core_eia930__hourly_balancing_authority_demand": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+    },
+    compute_kind="pandas",
+)
+def core_eia930__hourly_balancing_authority_assets(
+    raw_eia930__balance: pd.DataFrame,
+):
+    """Transforms raw_eia923__balance dataframe.
+
+    Extract the net generation information from the balance table and extract fuel type
+    from the net generation columns.
+    """
+    qual_cols = [
+        "report_datetime_local",
+        "report_datetime_utc",
+        "balancing_authority_code_eia",
+        # report_date
+        # eia_region_code,
+        # report_hour_local
+    ]
+    # Select only the columns relevant to the BA net generation table
+    netgen = raw_eia930__balance[
+        qual_cols
+        + list(raw_eia930__balance.filter(like="net_generation"))
+        + list(raw_eia930__balance.filter(like="interchange"))
+    ]
+
+    # Rename columns so that they contain only the energy source and the level of
+    # processing, so it's easy to construct a multi-index to unstack below
+    # Doing these manipulations while the values are in the column names rather than
+    # after they've been turned into a categorical column with millions of entries
+    # is much faster.
+    netgen_renamed = netgen.rename(
+        lambda col: col.removeprefix("net_generation_").removesuffix("_mw"),
+        axis="columns",
+    ).set_index(qual_cols)
+
+    # Prepare a multi-index for the columns so that we can stack cleanly
+    netgen_renamed.columns = pd.MultiIndex.from_tuples(
+        [x.split("_") for x in netgen_renamed.columns], names=["energy_source", None]
+    )
```

Comment:
it would be nice to add a description of the column format somewhere. without that in here this is a little hard to follow

Reply:
what happens with the interchange from the filtering of columns above in all of this? or does interchange end up as a energy source?

Reply:
The interchange has to be dealt with separately in the column filtering, but yes it ends up looking like an energy source because it also comes in "reported", "adjusted", and "imputed" flavors.